controller/volume_controller.go

```diff
@@ -250,8 +250,13 @@ func (vc *VolumeController) syncVolume(key string) (err error) {
 
 	log := getLoggerForVolume(vc.logger, volume)
 
+	defaultEngineImage, err := vc.ds.GetSettingValueExisted(types.SettingNameDefaultEngineImage)
+	if err != nil {
+		return err
+	}
+
 	if volume.Status.OwnerID != vc.controllerID {
-		if !vc.isResponsibleFor(volume) {
+		if !vc.isResponsibleFor(volume, defaultEngineImage) {
```

Comment:
It's better to move the isResponsible check outside the status.Owner != controller part.
See above PR comment, then move the below engineImage check into the isResponsible check.

Reply:
Thank you. Fixed